kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -25,231 +27,457 @@
 namespace Kratos
 {
 
-    FindIntersectedGeometricalObjectsProcess::FindIntersectedGeometricalObjectsProcess(ModelPart& rPart1, ModelPart& rPart2)
-        : mrModelPart1(rPart1), mrModelPart2(rPart2)
-    {
-    }
+template<class TEntity>
+FindIntersectedGeometricalObjectsProcess<TEntity>::FindIntersectedGeometricalObjectsProcess(
+    ModelPart& rPart1,
+    ModelPart& rPart2
+    ) : mrModelPart1(rPart1),
+        mrModelPart2(rPart2)
+{
+    const Parameters default_parameters = GetDefaultParameters();
+    mThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+FindIntersectedGeometricalObjectsProcess<TEntity>::FindIntersectedGeometricalObjectsProcess(
+    Model& rModel,
+    Parameters ThisParameters
+    ) : mrModelPart1(rModel.GetModelPart(ThisParameters["first_model_part_name"].GetString())),
```

Comment:
Maybe this is a good chance to use better names for the model parts. Even though the member variables keep their current names, I'd rename the Parameters constructor arguments to something more self-explaining like `intersected_model_part_name` and `intersecting_model_part_name`.

Reply:
I prefer to do this in a different PR, as I said, this is part #4263 and I want to avoid conflicts